terminus-core/src/components/tabHeader.component.pug

```diff
@@ -1,7 +1,7 @@
 .progressbar([style.width]='progress + "%"', *ngIf='progress != null')
-.index(
+.index(*ngIf='tab.showIndex==true',
     #handle,
     [style.background-color]='tab.color',
 ) {{index + 1}}
 .name([title]='tab.customTitle || tab.title') {{tab.customTitle || tab.title}}
-button((click)='app.closeTab(tab, true)') &times;
+button(*ngIf='tab.showCloseButton==true',(click)='app.closeTab(tab, true)') &times;
```

Comment:
You can make `config` `public` and use `!config.store.terminal.disableCloseButton` here directly